modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeVelocityVariableMigrationCheck.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.CharPool;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.VelocityMigrationUtil;
+
+import java.io.IOException;
+
+/**
+ * @author Nícolas Moura
+ */
+public class UpgradeVelocityVariableMigrationCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws IOException {
+
+		if (!fileName.endsWith(".vm")) {
+			return content;
+		}
+
+		VelocityMigrationUtil.writeMigratedContent(
+			_formatContent(
+				VelocityMigrationUtil.getFTLContentMigrated(content, fileName)),
+			fileName);
+
+		return content;
+	}
+
+	private String _formatContent(String content) {
+		String[] lines = content.split(StringPool.NEW_LINE);
+
+		for (String line : lines) {
+			if (line.contains("#set")) {
+				String newComment = line.replace("#set", "<#assign");
```

Comment:
@nicolas-sss, actually the name of this variable is related to the previous check implement by you, right?

I mean, the one related to migrate Velocity comments to FreeMarker

How about just rename it to `newContent` in order to reuse it across all other checks?

Reply:
Sorry about that, I will rename to `newVariable` to be valid for this context.